src/integration.spec.ts

```diff
@@ -235,4 +236,80 @@ describe('integration test with mongod', function () {
       }
     });
   });
+
+  context('can authenticate with a mock IdP - without id_token', function () {
+    let provider: OIDCMockProvider;
+    let connectionString: string;
+
+    before(async function () {
+      if (+process.version.slice(1).split('.')[0] < 16) {
+        // JWK support for Node.js KeyObject.export() is only Node.js 16+
+        // but the OIDCMockProvider implementation needs it.
+        return this.skip();
+      }
+      provider = await OIDCMockProvider.create({
+        getTokenPayload() {
+          return {
+            expires_in: 3600,
+            payload: {
+              // Define the user information stored inside the access tokens
+              groups: ['testgroup'],
+              sub: 'testuser',
+              aud: 'resource-server-audience-value',
+            },
+            // id_token will not be included
+            skipIdToken: true,
+          };
+        },
+      });
+
+      const serverOidcConfig = [
+        {
+          issuer: provider.issuer,
+          clientId: 'mockclientid',
+          requestScopes: ['mongodbGroups'],
+          authorizationClaim: 'groups',
+          audience: 'resource-server-audience-value',
+          authNamePrefix: 'dev',
+        },
+      ];
+
+      cluster = await spawnMongod(serverOidcConfig);
+      connectionString = `mongodb://${cluster.hostport}/?authMechanism=MONGODB-OIDC`;
+    });
+
+    after(async function () {
+      await Promise.all([cluster?.close?.(), provider?.close?.()]);
+    });
```

Comment:
This is just fine as it is, just an optional suggestion, you could deduplicate code here (and speed up tests a bit because it would not require spinning up and down a new `mongod` process) by merging it with the previous `context` block and assigning values to `getTokenPayload`, similar to what we do for `getTokenPayload` in https://github.com/mongodb-js/compass/blob/469affd924efe6ff9070fde101d8a13181af0dd6/packages/compass-e2e-tests/tests/oidc.test.ts#L166 (provide a default value for it in a `beforeEach()` hook, but individual tests can override it)

Reply:
good idea, thank you!